### PR TITLE
[fix][doc] fix many anchor errors

### PR DESCRIPTION
**Pull request by labuladong**
Master Issue: #18026

### Modifications

According to  https://github.com/apache/pulsar/issues/18026#issuecomment-1277110925, this pr fixes many markdown anchor errors.

For the link checker script and more errors see [#18026](https://github.com/apache/pulsar/issues/18026#issuecomment-1283652162)

### Documentation

- [x] `doc` 
- [ ] `doc-required` 
- [ ] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository: https://github.com/labuladong/pulsar/pull/3

### CODE REVIEW

File: site2/docs/admin-api-brokers.md [hunk 1]
```diff
@@ -147,7 +147,7 @@ admin.brokers().getOwnedNamespaces(cluster,brokerUrl);
 
 ### Dynamic broker configuration
 
-One way to configure a Pulsar [broker](reference-terminology.md#broker) is to supply a [configuration](reference-configuration.md#broker) when the broker is [started up](reference-cli-tools.md#pulsar-broker).
+One way to configure a Pulsar [broker](reference-terminology.md#broker) is to supply a [configuration](reference-configuration.md#broker) when the broker is [started up](reference-cli-tools.md).
```
Comment (RobertIndie) [inline site2/docs/admin-api-brokers.md:150]:
```suggestion
One way to configure a Pulsar [broker](reference-terminology.md#broker) is to supply a [configuration](reference-configuration.md#broker) when the broker is [started up](reference-cli-broker-tool.md).
```

File: site2/docs/administration-zk-bk.md [hunk 2]
```diff
@@ -122,7 +122,7 @@ peerType=observer
 
 ##### Start the service
 
-Once your configuration store configuration is in place, you can start up the service using [`pulsar-daemon`](reference-cli-tools.md#pulsar-daemon)
+Once your configuration store configuration is in place, you can start up the service using [`pulsar-daemon`](reference-cli-tools.md)
```
Comment (RobertIndie) [inline site2/docs/administration-zk-bk.md:125]:
```suggestion
Once your configuration store configuration is in place, you can start up the service using [`pulsar-daemon`](reference-cli-pulsar-daemon.md)
```

**Review by RobertIndie — COMMENTED** [review]
For the reference to `reference-cli-tools.md`, it's better to give the specific reference page for the user. Otherwise, it will confuse users. Here I give 2 examples.

File: site2/docs/admin-api-brokers.md (hunk 1 above)
Comment (labuladong) [reply site2/docs/admin-api-brokers.md:150]: Thanks for pointing this out, but I think this PR should focus on fixing the error anchors. We may improve the `reference-cli-tools.md` in another PR.

**Comment by Anonymitaet** [comment]
> For the reference to `reference-cli-tools.md`, it's better to give the specific reference page for the user. Otherwise, it will confuse users. Here I give 2 examples.

Generic links are preferred to reduce maintenance cost
Details see https://github.com/apache/pulsar/pull/18101#issuecomment-1283542571

**Comment by Anonymitaet** [comment]
@labuladong .md doc PR does not need this 
image

because: https://github.com/apache/pulsar/blob/912f344de571aba25809387ba8e71419d1f47b1b/.github/changes-filter.yaml#L9

**Review by Anonymitaet — APPROVED** [review]

File: site2/docs/admin-api-brokers.md (hunk 1 above)
Comment (RobertIndie) [reply site2/docs/admin-api-brokers.md:150]: Make sense.

**Review by RobertIndie — APPROVED** [review]

**Comment by labuladong** [comment]
@Anonymitaet @momo-jun Can you merge this pr? Thanks.

**Comment by labuladong** [comment]
@Anonymitaet @momo-jun Can you merge this pr? Thanks.

**Comment by Anonymitaet** [comment]
@labuladong This PR can not be merged because of the failed tests. Feel free to ping me if all tests pass.

**Comment by tisonkun** [comment]
@Anonymitaet I think now this PR is ready to merge.

**Pull request merged**